node/src/protocol/triple.rs

```diff
@@ -160,19 +155,8 @@ impl TripleManager {
     pub fn poke(&mut self) -> Result<Vec<(Participant, TripleMessage)>, ProtocolError> {
         let mut messages = Vec::new();
         let mut result = Ok(());
-        self.generators.retain(|id, generator| {
+        self.generators.retain(|id, protocol| {
             loop {
-                let mut protocol = match generator.write() {
-                    Ok(protocol) => protocol,
-                    Err(err) => {
-                        tracing::error!(
-                            ?err,
-                            "failed to acquire lock on triple generation protocol"
-                        );
-                        break false;
-                    }
-                };
-
                 let action = match protocol.poke() {
```

Comment:
Isn't this a read of the protocol?

Reply:
Nope, the read of the protocol comes from explicit `read` calls since it was behind a lock

Reply:
oh i see. protocol.read()